packages/atomic/src/components/commerce/atomic-commerce-query-summary/atomic-commerce-query-summary.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  ListingSummary,
+  ListingSummaryState,
+  Pagination,
+  SearchSummary,
+  SearchSummaryState,
+  buildListingSummary,
+  buildSearchSummary,
+} from '@coveo/headless/commerce';
+import {Component, h, State} from '@stencil/core';
+import {AriaLiveRegion} from '../../../utils/accessibility-utils';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {LocalizedString} from '../../../utils/jsx-utils';
+import {QuerySummaryContainer} from '../../common/query-summary/container';
+import {QuerySummaryGuard} from '../../common/query-summary/guard';
+import {getQuerySummaryI18nParameters} from '../../common/query-summary/utils';
+import {CommerceBindings} from '../atomic-commerce-interface/atomic-commerce-interface';
+
+/**
+ * The `atomic-commerce-query-summary` component displays information about the current range of results and the request duration (e.g., "Results 1-10 of 123 in 0.47 seconds").
+ *
+ * @part container - The container for the whole summary.
+ * @part results - The container for the results.
+ * @part duration - The container for the duration.
+ * @part highlight - The summary highlights.
+ * @part query - The summary highlighted query.
+ * @part placeholder - The query summary placeholder used while the search interface is initializing.
+ *
+ * @internal
+ */
+@Component({
+  tag: 'atomic-commerce-query-summary',
+  styleUrl: 'atomic-commerce-query-summary.pcss',
+  shadow: true,
+})
+export class AtomicQuerySummary
+  implements InitializableComponent<CommerceBindings>
+{
+  @InitializeBindings() public bindings!: CommerceBindings;
+  public listingOrSearchSummary!: SearchSummary | ListingSummary;
+  public pagination!: Pagination;
+
+  @BindStateToController('listingOrSearchSummary')
+  @State()
+  private listingOrSearchSummaryState!:
+    | SearchSummaryState
+    | ListingSummaryState;
+
+  @State() public error!: Error;
+
+  @AriaLiveRegion('query-summary')
+  protected ariaMessage!: string;
+
+  public initialize() {
+    if (this.bindings.interfaceElement.type === 'product-listing') {
+      this.listingOrSearchSummary = buildListingSummary(this.bindings.engine);
+    } else {
+      this.listingOrSearchSummary = buildSearchSummary(this.bindings.engine);
+    }
+  }
+
+  public render() {
+    const {
+      firstProduct,
+      firstSearchExecuted,
+      lastProduct,
+      totalNumberOfProducts,
+      hasProducts,
+      hasError,
+    } = this.listingOrSearchSummaryState;
+
+    const {i18nKey, highlights, ariaLiveMessage} =
+      getQuerySummaryI18nParameters({
+        first: firstProduct,
+        last: lastProduct,
+        query:
+          'query' in this.listingOrSearchSummaryState
+            ? this.listingOrSearchSummaryState.query
+            : '',
```

Comment:
If we don't want/care about null or undefined values, I'd simplify it like this
```suggestion
        query: this.listingOrSearchSummaryState.query ?? '',
```
are we sure also that query can be undefined/null/missing?

Reply:
> are we sure also that query can be undefined/null/missing?

Yes, since it's a `listingOrSearchState` (2 types, components works for both product listing page or search page), and it's expected that there's no query for `productListing`